parser/parser.go

```diff
@@ -38,14 +39,14 @@ func (p *Parser) Parse(kubeFile []byte, name string) *v1.Image {
 			logger.Warn("decode kubeFile line failed %v", err)
 			continue
 		}
-
-		//TODO count layer hash
 		image.Spec.Layers = append(image.Spec.Layers, v1.Layer{
 			Hash:  "",
 			Type:  layerType,
 			Value: layerValue,
 		})
 	}
+	// gen consistent image ID
+	image.Spec.ID = utils.MD5(kubeFile, []byte(name))
```

Comment:
curious why generate id base on kubefile and image name instead of utils.GenUniqueID(32)

Reply:
This is not appropriate and may even lead to bugs if two different image using same Clusterfile. Image ID should not using Clusterfile MD5

Reply:
like rootfs image. it can be affect only the kubernetes version. not the others things.

Reply:
image id = md5sum(clusterfile+image name), same clusterfile with same image name guess that is a same image . what do you think ?@all

Reply:
Not good..

Reply:
docker generate image id refer to content of image config. Is it fine to calculate image id depends on image metadata?